u8timeseries/timeseries.py

```diff
@@ -804,10 +815,41 @@ def _combine_from_pd_ops(self, other: 'TimeSeries',
         conf_hi = self._combine_or_none(self._confidence_hi, other.conf_hi_pd_series(), combine_fn)
         return TimeSeries(series, conf_lo, conf_hi)
 
+    @staticmethod
+    def _fill_missing_dates(series: pd.Series) -> pd.Series:
+        """
+        Tries to fill missing dates in series with NaN.
+        Method is successful only when explicit frequency can be determined from all consecutive triple timestamps.
+
+        Parameters
+        ----------
+        series
+            The actual time series, as a pandas Series with a proper time index.
+
+        Returns
+        -------
+        pandas.Series
+            A new Pandas Series without missing dates.
+        """
+        date_axis = series.index
+        samples_size = 3
+        observed_frequencies = [
+            date_axis[x:x + samples_size].inferred_freq
```

Comment:
How does this inferred_freq work internally?

We take a chunk of 3 samples and try to infer frequency, then take another 3 from x+1. Could you make sure this is not O(N^2)?

Reply:
Looks like linear, because it works on deltas between points, so at the end it is linear.